xen_hypervisor: Treat missing socket as temporary failure

This socket file is created by xenfs, which can be a module. If that
is the case the hypervisor can be initialized later.

Index: libvirt-0.9.13/src/xen/xen_hypervisor.c
===================================================================
--- libvirt-0.9.13.orig/src/xen/xen_hypervisor.c	2012-08-15 12:45:58.213198574 +0200
+++ libvirt-0.9.13/src/xen/xen_hypervisor.c	2012-08-15 12:46:02.981221881 +0200
@@ -2034,6 +2034,9 @@
     ret = open(XEN_HYPERVISOR_SOCKET, O_RDWR);
     if (ret < 0) {
         hv_versions.hypervisor = -1;
+        in_init = 0;
+        /* Missing socket may appear after xenfs is loaded as a module */
+        initialized = 0;
         return -1;
     }
     fd = ret;
